service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt

```diff
@@ -291,9 +291,38 @@ WITH placement_ids AS (
         WHERE id = :accountId
     )
     AND Daterange(pl.start_date, pl.end_date, '[]') @> current_date
+    AND NOT (
+        SELECT EXISTS (
+            SELECT 1
+            FROM messaging_blocklist
+            WHERE child_id = pl.child_id
+            AND blocked_recipient = guardian_id
+            )
+        )
+
+    UNION
+
+    SELECT pl.id AS id
+    FROM fridge_child fg
+    JOIN placement pl
+    ON fg.child_id = pl.child_id
+    WHERE head_of_child = (
+        SELECT person_id AS id
+        FROM message_account
+        WHERE id = :accountId
+    )
+    AND Daterange(pl.start_date, pl.end_date, '[]') @> current_date
```

Comment:
should also have `AND Daterange(fg.start_date, fg.end_date, '[]') @> current_date AND fg.conflict = false`